front/src/modules/scenario/components/AddOrEditScenarioModal.tsx

```diff
@@ -243,18 +244,24 @@ export default function AddOrEditScenarioModal({
             {!editionMode && electricalProfilOptions.length > 1 && (
               <div className="scenario-edition-modal-description">
                 <SelectImprovedSNCF
-                  title={
+                  label={
                     <div className="d-flex align-items-center">
                       <span className="mr-2">
                         <GiElectric />
                       </span>
                       {t('electricalProfileSet')}
                     </div>
                   }
-                  selectedValue={selectedValue}
-                  options={electricalProfilOptions}
-                  onChange={(e: SelectOptionsType) =>
-                    setCurrentScenario({ ...currentScenario, electrical_profile_set_id: e.key })
+                  value={selectedValue}
+                  options={electricalProfilOptions.map((e) => ({
+                    id: `${e.key}`,
+                    label: e.value,
+                  }))}
```

Comment:
I think it's better to change the electricalProfilOptions initialization, line 63:
```
    ({
        ...response,
        electricalProfilOptions: [
          noElectricalProfileSetOption,
          ...sortBy(response.data, ['name']).map((option: ElectricalProfileSetType) => ({
            id: `${option.id}`,
            label: option.name,
          })),
        ],
      }),
```

and then 
```suggestion
                  options={electricalProfilOptions}
```

Reply:
We have an other issue here : the select takes only string for ID, but in `electricalProfilOptions` it's a number. , so we need to cast it (on the onchange I do a `+e.id`). for the select component.

If I do it directly when we initialize it, we lost its primary type (same for the selectedValue)., which can leads to bad type if someone want to use it.

The type string is only needed for the select component, that's the cast is only done on it..